src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1065,106 +1193,140 @@ private void ImportBinaryOperation(ILOpcode opcode)
             }
 
             // The one exception from the above rule
-            if ((kind == StackValueKind.ByRef) &&
-                    (opcode == ILOpcode.sub || opcode == ILOpcode.sub_ovf || opcode == ILOpcode.sub_ovf_un))
+            if (kind == StackValueKind.ByRef || kind == StackValueKind.NativeInt)
```

Comment:
This makes sense for ByRef, but a NativeInt shouldn't need to be made into a pointer